engine/src/main/java/org/pentaho/di/trans/DefaultTransFactoryManager.java

```diff
@@ -0,0 +1,59 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.di.trans;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The DefaultTransManager is used to manage the registration of TransFactory and
+ * generate the Trans Object for respective RunConfiguration
+ */
+public class DefaultTransFactoryManager implements TransFactoryMananger {
+
+  private static DefaultTransFactoryManager instance;
+
+  private Map<String, TransFactory> mapConfigurationNameTransFactory;
+
+
+  private DefaultTransFactoryManager(){
+    mapConfigurationNameTransFactory = new HashMap<>();
+  }
+
+  public static synchronized DefaultTransFactoryManager getInstance() {
+    if ( null == instance ) {
+      instance = new DefaultTransFactoryManager();
+    }
+    return instance;
+  }
+
+  /**
+   * Registers/Adds a TransFactory using the provided Run Configuration Name
+   * @param runConfigurationName
+   * @param transFactory
+   */
+  @Override
+  public void registerFactory( String runConfigurationName, TransFactory transFactory ) {
+    mapConfigurationNameTransFactory.put( runConfigurationName, transFactory );
+  }
+
+  /**
+   * Fetches a TransFactory or Returns default TransFactoryImpl using the provided Run Configuration Name
```

Comment:
javadoc comment references old class name  "TransFactoryImpl" instead of new name
You should use @see @link so the javadoc will have a hyperlink or tooltip to the actual class - https://www.baeldung.com/javadoc-see-vs-link#2-displaying-see-and-linktag
and any rename changes or refactor, an IDE will automatically pick it up.

Reply:
addressed